### PR TITLE
Transactions submitted by sendrawtransaction rpccall should be always allowed despite this new -relaypriority switch.

### PR DESCRIPTION
**Pull request by wangchun**
Transactions submitted by sendrawtransaction rpccall should be always allowed despite this new -relaypriority switch.

Allow duplicate txout address in createrawtransaction rpccall.

Added microsecond for logtimestamps.

Tab indent fix.

### CODE REVIEW

File: src/netbase.cpp [hunk 1]
```diff
@@ -788,15 +788,15 @@ bool CNetAddr::IsTor() const
 bool CNetAddr::IsLocal() const
 {
     // IPv4 loopback
-   if (IsIPv4() && (GetByte(3) == 127 || GetByte(3) == 0))
-       return true;
+    if (IsIPv4() && (GetByte(3) == 127 || GetByte(3) == 0))
+        return true;
```
Comment (jonasschnelli) [inline src/netbase.cpp]: Why did you change this lines?

File: src/netbase.cpp [hunk 2]
```diff
@@ -788,15 +788,15 @@ bool CNetAddr::IsTor() const
...
-   static const unsigned char pchLocal[16] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
-   if (memcmp(ip, pchLocal, 16) == 0)
-       return true;
+    // IPv6 loopback (::1/128)
+    static const unsigned char pchLocal[16] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
+    if (memcmp(ip, pchLocal, 16) == 0)
+        return true;
 
-   return false;
+    return false;
```
Comment (jonasschnelli) [inline src/netbase.cpp]: Why did you change netbase.cpp at all?

File: src/rpcmining.cpp [hunk 3]
```diff
@@ -86,7 +86,7 @@ UniValue getnetworkhashps(const UniValue& params, bool fHelp)
             "\nExamples:\n"
             + HelpExampleCli("getnetworkhashps", "")
             + HelpExampleRpc("getnetworkhashps", "")
-       );
+        );
```
Comment (jonasschnelli) [inline src/rpcmining.cpp]: Same for this file.

**Comment by jonasschnelli** [comment]
I think only your change in `main.cpp` line 1003 is relevant.
You should probably remove all other changes and force-push again.

**Comment by jonasschnelli** [comment]
I think the concept of allowing local wallet `send*` commands to always end up in the memory pool is not a undeliberated idea. It didn't thought it through completely but somehow i agree that local wallet send commands should always end up in the local memory pool.
Sorry for that, but the code is ugly and unusable.
NACK for the code but idea seems usefull

File: src/main.cpp [hunk 4]
```diff
@@ -1000,7 +1000,7 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
                              REJECT_INSUFFICIENTFEE, "insufficient fee");
 
         // Require that free transactions have sufficient priority to be mined in the next block.
-        if (GetBoolArg("-relaypriority", true) && nFees < ::minRelayTxFee.GetFee(nSize) && !AllowFree(view.GetPriority(tx, chainActive.Height() + 1))) {
+        if (fLimitFree && GetBoolArg("-relaypriority", true) && nFees < ::minRelayTxFee.GetFee(nSize) && !AllowFree(view.GetPriority(tx, chainActive.Height() + 1))) {
```
Comment (jonasschnelli) [inline src/main.cpp]: I think you misuse `fLimitFree` here.

File: src/netbase.cpp (hunk 1 above)
Comment (wangchun) [inline src/netbase.cpp]: These lines do not follow the code style by only having three space characters for a tab indent.

Comment (jonasschnelli) [inline src/netbase.cpp]: Okay. I see. You are right that the formatting is wrong there. But the diff you create can be hard to read for others. At least it would be good to separate the style changes from the actual code changes. Also i wouldn't care to much about existing inconsistency in code formatting because we could run one `clang-format` over everything (see https://github.com/bitcoin/bitcoin/pull/5387).

File: src/util.cpp [hunk 5]
```diff
@@ -243,8 +244,13 @@ int LogPrintStr(const std::string &str)
...
         // Debug print useful for profiling
-        if (fLogTimestamps && fStartedNewLine)
-            ret += fprintf(fileout, "%s ", DateTimeStrFormat("%Y-%m-%d %H:%M:%S", GetTime()).c_str());
+        if (fLogTimestamps && fStartedNewLine) {
+            std::locale loc(std::locale::classic(), new boost::posix_time::time_facet("%Y-%m-%dT%H:%M:%sZ"));
+            std::stringstream ss;
+            ss.imbue(loc);
+            ss << boost::posix_time::microsec_clock::universal_time();
+            ret += fprintf(fileout, "[%s] ", ss.str().c_str());
+        }
```
Comment (jonasschnelli) [inline src/util.cpp]: What is the reason for this? Is this required to achieve your PRs goal or is this another feature that is unrelated to this PR?

**Comment by laanwj** [comment]
To make it easier to discuss specific changes. It's better to submit unrelated changes separately:

- Transactions submitted by sendrawtransaction rpccall should be always allowed despite this new -relaypriority switch: this may be slightly controversial - we've always have the implicit policy 'don't send transactions that you wouldn't relay yourself'

- Allow duplicate txout address in createrawtransaction rpccall: fair enough

- Added microsecond for logtimestamps: this is a feature unrelated to your RPC changes

- Tab indent fix: whitespace 'improvents', if you do them at all, should go to @theuni's trivial-next branch:  https://github.com/theuni/bitcoin

**Comment by luke-jr** [comment]
I don't see a reason to allow address reuse in createrawtransaction.

**Comment by sipa** [comment]
Closing until @laanwj's comments are addressed: this PR tries to do many things at once, making it hard to review.
